feat: support DeviceRequests (service.gpus and reservations.devices)

jabr opened the pull request:
Update: I'm successfully running an uncloud service with GPU access using a custom `uc`/`uncloudd` of this branch.

New tests pass and it compiles for me ~, but I have not yet tried this on a real deploy~ .

jabr commented:
I'm now running this build of `uncloudd` on my GPU server and successfully deployed a test service with this `uc` build:
```
services:
  nvidia:
    image: nvidia/cuda:12.8.1-base-ubuntu24.04
    command: tail -f /dev/null
    x-machines: [ gpu-server ]
    gpus: all
```

I then verified it has access to the GPU with (run on `gpu-server`):
```
docker exec <container-id-of-nvidia-service> nvidia-smi
```

psviderski commented on pkg/api/resources.go:
```diff
@@ -13,4 +17,6 @@ type ContainerResources struct {
 	// TODO: implement a placement constraint that checks available memory on machines.
 	MemoryReservation int64
+	// DeviceRequests for access to things like GPUs
+	DeviceRequests []container.DeviceRequest
```
Just a tiny nit, regardless of the name of the docker internal struct, let's call it `DeviceReservations` to be consistent with the compose terminology and config (limits and reservations) and the `MemoryReservation` above.

jabr commented:
Just a heads up that I apparently broke something in the last cleanup commit or two on this. The DeviceRequests aren't getting sent in the spec to the servers anymore so a subsequent redeploy of a service didn't have GPU access.

I'll take a look shortly. Apologies for not testing it again after the initial commit. 😢

jabr commented:
Ah, nevermind, false alarm.

I still had the first version of the daemon running on my server. It was still expecting the `DeviceRequests` name that we changed to `DeviceReservations`.

I updated both it and my local `uc` to a fresh build from `main` and it's working fine.
